tests/test_pipeline_output.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+import numpy as np
+import os
+import allensdk.core.json_utilities as ju
+
+
+import sys
+sys.path.append("/local1/pyprojects/allensdk.ipfx/bin") # there is got to be a better way
+from run_pipeline import run_pipeline
+
+
+def check_json_files_are_equal(benchmark_json,test_json):
+
+    benchmark_pipeline_output = ju.read(benchmark_json)
+    test_pipeline_output = ju.read(test_json)
+
+    return benchmark_pipeline_output == test_pipeline_output
+
+
+def test_pipeline_output():
+    """
+    Validates pipeline output for a list of specimens.
+    Runs pipeline, saves data to disk and compares pipeline output to that from the benchmark runs.
+    If pipeline output has changed (was not validated), raises assertion error and lists specimens failing validation.
+
+
+    """
+    specimens = np.array([500844783, 509604672, 513876493])
+
+    validation = []
+
+    for specimen in specimens:
+        print("testing specimen %s" % specimen)
+        benchmark_dir = "/allen/aibs/informatics/module_test_data/ipfx/benchmark_pipeline_output/%s" % specimen
+        test_dir = "/allen/aibs/informatics/module_test_data/ipfx/test_pipeline_output/%s" % specimen
```

Comment:
Use `tmpdir_factory` for this.  You can see how @NileGraddis did it here:
https://github.com/AllenInstitute/AllenSDK/blob/b2d0bac25a9e2b6a587be9d7902a25d0e6bfe936/allensdk/test/test_inline_examples.py#L20

the `tmpdir_factory` object allows you to create a temporary directory and by default these will go into `tmp` that will get cleaned up.  However, you can override the default temp directory from the command line like this:

`pytest --basetemp=some_dir`

https://docs.pytest.org/en/latest/tmpdir.html

So bamboo can keep the files or not.

Reply:
I expect this will look something like this:
```python
@pytest.mark.parametrize('specimen_id,test_pipeline_input', zip(TEST_PIPELINE_INPUTS, TEST_SPECIMEN_IDS))
def test_pipeline_output(specimen_id, test_pipeline_input, tmpdir_factory): # tmpdir_factory should just exist
   #...
   test_pipeline_output = tmpdir_factory.mktemp("data").join('%d_pipeline_output.json' % specimen_id)
   # ...
```